static/app/utils/useMetricMetas.tsx

```diff
@@ -1,31 +1,15 @@
-import {useEffect} from 'react';
+import {useContext} from 'react';
 
-import {fetchMetricsFields} from 'sentry/actionCreators/metrics';
-import MetricsMetaStore from 'sentry/stores/metricsMetaStore';
-import PageFiltersStore from 'sentry/stores/pageFiltersStore';
-import {useLegacyStore} from 'sentry/stores/useLegacyStore';
-import useApi from 'sentry/utils/useApi';
-
-import useOrganization from './useOrganization';
+import {MetricsContext} from 'sentry/utils/metrics/metricsContext';
 
 export function useMetricMetas() {
-  const api = useApi();
-  const organization = useOrganization();
-  const {selection} = useLegacyStore(PageFiltersStore);
-  const {metricsMeta, loaded} = useLegacyStore(MetricsMetaStore);
-  const shouldFetchMetricsMeta = !loaded;
-
-  useEffect(() => {
-    let unmounted = false;
-
-    if (!unmounted && shouldFetchMetricsMeta) {
-      fetchMetricsFields(api, organization.slug, selection.projects);
-    }
+  const metricsContext = useContext(MetricsContext);
 
-    return () => {
-      unmounted = true;
-    };
-  }, [selection.projects, organization.slug, shouldFetchMetricsMeta]);
+  if (!metricsContext) {
+    throw new Error('useMetricTags was called outside of MetricsContextProvider');
```

Comment:
```suggestion
    throw new Error('useMetricsMeta was called outside of MetricsContextProvider');
```

Reply:
oh, I think Jonas updated this by mistake when he pushed a commit... thanks Shruti